src/com/company/tests/BoardTest.java

```diff
@@ -0,0 +1,62 @@
+package com.company.tests;
+
+import com.company.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+public class BoardTest {
```

Comment:
Brakuje mi tutaj jeszcze paru testów.
Np: 
Czy da się wstawić `X` a potem w to samo pole `O` ?
Czy listener wywoła odpowiednią metodę kiedy ktoś wygra?
Czy listener wywoła odpowiednią metodę kiedy nikt nie wygra?

Reply:
hmm...
Czy da się wstawić X a potem w to samo pole O ? -> getSignTextTestXchangedToO
Czy listener wywoła odpowiednią metodę kiedy ktoś wygra? -> TestListenerWhenWin() , niestety przy wywołaniu dostaję błąd którego nie rozumiem, java.lang.NullPointerException w linii która odpowiada za wstawienie ostatniego znaku do tablicy. Sprawdź proszę co zrobiłam.
Czy listener wywoła odpowiednią metodę kiedy nikt nie wygra? - przy tablicy 2x2 się nie da wypełnić tablicy tak żeby nikt nie wygrał, spróbuję jutro pomyśleć jak to zrobić.

Reply:
Co do null pointera to przekazujesz nulla zamiast endGameListenera i kiedy dochodzi do mementu w którym gra próbuje go użyć to ma problem bo się nie da.
Musisz: 
1. Zabezpieczyć swoją grę żeby w takim przypadku się nie wywalała
2. Nie przekazywać listenera do konstruktora

Jak się nie da to zmień rozmiar tablicy do 3 i już :)